lib/util/string_conv/string_conv.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <logging/log.h>
+
+#define LOG_MODULE_NAME string_conv
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_STRING_CONV);
+
+int string_conv_str2long(char *str, int *val)
+{
+	int str_len = strlen(str);
+	int temp_val;
+	int idx = 0;
+
+	if (!strncmp(str, "-", 1)) {
+		idx = 1;
+	}
+
+	for (int i = idx; i < str_len; i++) {
+		if (isdigit((int)str[i])) {
+			continue;
+		}
+
+		LOG_ERR("\"%s\" is not recognized as a valid signed decimal input", str);
+		return -EINVAL;
+	}
+
+	temp_val = strtol(str, NULL, 10);
+
+	if ((temp_val <= LONG_MIN) || (temp_val >= LONG_MAX)) {
```

Comment:
`temp_val` is `int` `strtol` returns `long`, there is a chance that the if does not work as expected because you will already have value truncated here by implicit cast that happens on assignment to `temp_val`.